### ISSUE TITLE
[FEATURE REQUEST] Archiving "SpeciesConc" in molec/cm3 units in a separate diagnostic

### ISSUE THREAD

**Issue by jimmielin** [geoschem/geos-chem#1022]
### Request a new GEOS-Chem feature:

Hi all,

During the validation of the upcoming auto-reduction feature in KPP (within GEOS-Chem) I've ran into the need of converting the default `SpeciesConc` from `v/v dry`, the long-standing unit for output, into `molec/cm3`.

While this could technically be done outside of GEOS-Chem in a diagnostics package, perhaps `Convert_Spc_Units` in `UnitConv_Mod` could be leveraged within GEOS-Chem to do this more easily.

I'm wondering whether there is interest for such a feature to be added to GEOS-Chem mainline. If yes I'm happy to submit a pull request for the code I have in this commit: https://github.com/jimmielin/geos-chem/commit/b1599c98d06fbd0e5d4064e1cc34dbcc51abc3b8 which adds a `SpeciesConcMND_?ADV?` diagnostic (molecular number density; naming is to be consistent with the naming of the conversion routines in `unitconv_mod.F90`, which is `ConvertSpc_KgKgDry_to_MND`)

*P.S.: Not exactly off-topic but: https://twitter.com/ChemDelphine/status/1454152636492500995?s=20 - what we refer to `SpeciesConc` in GEOS-Chem (v/v dry) is actually volume mixing ratio or mole fraction. So if alternate units are added the naming could get confusing...*

Thanks!
Haipeng

**Comment by lizziel** [comment]
Daniel Jacob (@djxjacob) wrote:
```
Are you suggesting an additional diagnostic in GEOS-Chem for users to produce output in units of molecules cm-3? That would be useful – some variables (like OH) are more commonly presented as molecules cm-3, and aerosol concentrations are commonly presented as ug m-3 which users could readily convert from molecules cm-3. 

This made me think of a basic question – how do users presently convert aerosol output from v/v to the common unit of ug m-3?  Do we provide molecular weights for that purpose? I’m thinking in particular of dust which does not have a well-defined molecular weight, but for which we have to assign a molecular weight in GEOS-Chem.
```

**Comment by lizziel** [comment]
I think this would be a great update to add to 14.0 which will be released after 13.4. For naming, how about changing `SpeciesConc_` to the following two options: `SpeciesVMR_` and `SpeciesMND_`? This would go well with the other species update in 14.0 where we switch to`State_Chm%Species(N)%Conc(I,J,L)` for accessing concentrations in the model. This new diagnostic naming would distance itself from the generic (and sometimes wrong) use of `Conc` for concentration, and keep that reserved for time-varying units.

I am pinging @jennyfisher on this since she had an interest in the naming of the new species container internal to the model, and that interest may extend to the diagnostics too.

Regarding post-processing conversion to ug m-3, users can get molecular weights from the species database.

**Comment by jimmielin** [comment]
Sounds good. I think that having `SpeciesVMR_` and `SpeciesMND_` as the prefix names make more sense than `SpeciesConc`. It would be clear for users when they want to choose one in `HISTORY.rc`.

Speaking about 14.0.0 this would also mean that the auto-reduction capability for KPP (which @msl3v and @yantosca and I are working on) would go in a 14.x release? Species array changes, KPP solver optimizations, etc. would be a good fit for a new major version.

Thanks!

**Comment by djxjacob** [comment]
About moving to 14.0 after 13.4 ….One thing we heard from the GCSC last week is that MEGAN3 and the associated canopy physics should be moved into the ‘feature integration in the queue’ table (please do so) and be implemented in the 13 series before going to 14.0. Can we start working on it to have it ready for 13.4.0?   Daniel

**Comment by yantosca** [comment]
@djxjacob, we can make a separate feature request for MEGAN3.  The only thing is that we do not as yet have a clean PR but just code that will need to be manually updated (which will take time).  I've updated the wiki accordingly.

Also MEGAN3 would be a HEMCO update and less affected by the 14.0 species array modifications (I think).

**Comment by msulprizio** [comment]
We will not have time to include this in 14.0.0, so I have removed that milestone. We can likely add this in 14.1.0. Note that changing the name of `SpeciesConc_` to `SpeciesVMR_` and `SpeciesMND_` would require updates to GCPy's benchmarking scripts.

@jimmielin Could you submit your PR for the `SpeciesMND_` update at your earliest convenience so we can make sure to include it in the next version?

**Comment by jimmielin** [comment]
Hi Melissa, please find the pull request here - https://github.com/geoschem/geos-chem/pull/1264. Thank you!

**Comment by msulprizio** [comment]
This update was included in 14.1.0.
